src/librustc_middle/ty/context.rs

```diff
@@ -2612,6 +2613,22 @@ impl<'tcx> TyCtxt<'tcx> {
     }
 }
 
+impl TyCtxtAt<'tcx> {
+    /// Constructs a `TyKind::Error` type and registers a `delay_span_bug` to ensure it gets used.
+    #[track_caller]
+    pub fn ty_error(self) -> Ty<'tcx> {
+        self.tcx.ty_error_with_message(self.span, "TyKind::Error constructed but no error reported")
+    }
+
+    /// Constructs a `TyKind::Error` type and registers a `delay_span_bug` with the given `msg to
+    /// ensure it gets used.
+    #[track_caller]
+    pub fn ty_error_with_message(self, msg: &str) -> Ty<'tcx> {
+        self.sess.delay_span_bug(self.span, msg);
+        self.mk_ty(Error(super::sty::DelaySpanBugEmitted(())))
```

Comment:
```suggestion
        self.tcx.ty_error_with_message(self.span, msg)
```

Perhaps I'm missing something?

Reply:
No I'm just dumb :P

Reply:
This change doesn't seem to have been made?

Reply:
This change still hasn't been made?